lib/core/topologies/replset.js

```diff
@@ -277,15 +277,26 @@ function rexecuteOperations(self) {
 }
 
 function connectNewServers(self, servers, callback) {
+  // No new servers
+  if (servers.length === 0) {
+    return callback();
+  }
+
   // Count lefts
   var count = servers.length;
   var error = null;
 
+  function done() {
+    count = count - 1;
+    if (count === 0) {
+      callback(error);
+    }
+  }
+
   // Handle events
   var _handleEvent = function(self, event) {
     return function(err) {
       var _self = this;
-      count = count - 1;
 
       // Destroyed
       if (self.state === DESTROYED || self.state === UNREFERENCED) {
```

Comment:
This will short-circuit the callback and return without calling `done` (which used to happen) and without decrementing the count (which didn't used to happen). Is that intended?

Reply:
nice catch, fixed